src/value.rs

```diff
@@ -42,46 +42,82 @@ unsafe impl FromValue for Value {
     }
 }
 
-const NONE: Value = Value(sys::NONE);
-const UNIT: Value = Value(sys::UNIT);
+unsafe impl<'a, T> IntoValue for OCaml<'a, T> {
+    fn into_value(self, _rt: &Runtime) -> Value {
+        unsafe { Value::new(self.raw()) }
+    }
+}
+
+/*unsafe impl<'a, T> FromValue for OCaml<'a, T> {
+    fn from_value(value: Value) -> OCaml<'a, T> {
+        let rt = unsafe { &Runtime::recover_handle() };
+        let x: OCaml<T> = unsafe { OCaml::new(rt, value.0) };
+        unsafe { core::mem::transmute(x) }
+    }
+}*/
+
+unsafe impl<'a, T> IntoValue for OCamlRef<'a, T> {
+    fn into_value(self, _rt: &Runtime) -> Value {
+        unsafe { Value::new(self.get_raw()) }
+    }
+}
+
+unsafe impl crate::interop::ToOCaml<Value> for Value {
+    fn to_ocaml<'a>(&self, gc: &'a mut Runtime) -> OCaml<'a, Value> {
+        unsafe { OCaml::new(gc, self.0) }
+    }
+}
+
+unsafe impl<'a> crate::interop::FromOCaml<Value> for Value {
+    fn from_ocaml(v: OCaml<Value>) -> Value {
+        unsafe { Value::new(v.raw()) }
+    }
+}
+
+const NONE: Value = unsafe { Value::new(sys::NONE) };
+const UNIT: Value = unsafe { Value::new(sys::UNIT) };
 
 impl Value {
-    /// Returns a named value registered by OCaml
-    pub fn named<T: FromValue>(name: &str) -> Option<T> {
-        unsafe {
-            let s = match crate::util::CString::new(name) {
-                Ok(s) => s,
-                Err(_) => return None,
-            };
-            let named = sys::caml_named_value(s.as_ptr());
-            if named.is_null() {
-                return None;
-            }
+    /// Convert from `Value` into `interop::OCaml<T>`
+    pub fn interop<'a, T>(&self, rt: &'a Runtime) -> OCaml<'a, T> {
```

Comment:
This one probably needs to be marked `unsafe`, because as it is it allows you to convert into any `T`, which is not always valid.